api/recompress_chunk.md

```diff
@@ -41,49 +40,26 @@ chunk for the first time, use [`compress_chunk`](/api/latest/compression/compres
 |-|-|-|
 |`chunk`|`REGCLASS`|The chunk to be recompressed. Must include the schema, for example `_timescaledb_internal`, if it is not in the search path.|
 
-## Optional arguments
-
-|Name|Type|Description|
-|-|-|-|
-|`if_not_compressed`|`BOOLEAN`|If `true`, prints a notice instead of erroring if the chunk is already compressed. Defaults to `false`.|
-
 ## Sample usage
 
 Recompress the chunk `timescaledb_internal._hyper_1_2_chunk`:
 
 ```sql
-CALL recompress_chunk('_timescaledb_internal._hyper_1_2_chunk');
+SELECT recompress_chunk('_timescaledb_internal._hyper_1_2_chunk');
```

Comment:
Is this a function again?

If so, then splitting the decompression and subsequent recompression into separate transaction should not be possible, and we might end up in unexpected deadlocks.

Reply:
Well, not very concerned about that we are not discussing locking, but we turned it into a procedure to be able to split up the decompression phase and compression phase into two transactions to avoid deadlocks. Has this changed?

Reply:
The recompress_chunk stuff in the PR is all outdated. The old procedure is now deprecated and will call compress_chunk. There is no new recompress_chunk function. And we do require less heavy locks now due to compressed chunks no longer being in an inheritance tree.

Reply:
But it can still lead to deadlocks, right? For both compression and decompression you need to truncate a table, which requires a heavy lock, and two heavy locks are sufficient to create a deadlock in some odd situation.